pkg/download/downloader.go

```diff
@@ -58,15 +64,41 @@ func Download(d Downloader) (int, io.ReadCloser, error) {
 		return resp.StatusCode, resp.Body, nil
 	}
 
-	err = fmt.Errorf("unexpected status code: actual=%d expected=%d", resp.StatusCode, http.StatusOK)
+	errString := ""
+	requestId := resp.Header.Get(xMsServiceRequestIdHeaderName)
 	switch d.(type) {
 	case *blobWithMsiToken:
 		switch resp.StatusCode {
 		case http.StatusNotFound:
-			return resp.StatusCode, nil, errors.Wrapf(err, MsiDownload404ErrorString)
+			errString = MsiDownload404ErrorString
 		case http.StatusForbidden:
-			return resp.StatusCode, nil, errors.Wrapf(err, MsiDownload403ErrorString)
+			errString = MsiDownload403ErrorString
+		}
+		break
+	default:
+		switch resp.StatusCode {
+		case http.StatusUnauthorized:
+			errString = fmt.Sprintf("CustomScript failed to download the blob [REDACTED] because access was denied. Please fix the blob permissions and try again, the response code returned was: %q",
+				resp.Status,
+			)
+		case http.StatusNotFound:
+			errString = fmt.Sprintf("CustomScript failed to download the blob [REDACTED] because it does not exist. Please create the blob and try again, the response code returned was %q",
+				resp.Status)
+
+		case http.StatusBadRequest:
+			errString = fmt.Sprintf("CustomScript failed to download the blob [REDACTED] because parts of the request were incorrectly formatted, missing, and/or invalid. The response code returned was: %q",
+				resp.Status)
+
+		case http.StatusInternalServerError:
+			errString = fmt.Sprintf("CustomScript failed to download the blob [REDACTED] due to an issue with storage. The response code returned was: %q",
```

Comment:
Is it possible to include the error message returned from the downstream service like storage if it is available in response ? That'd help us identify issue faster.

Reply:
Yes, the response code and message are both in resp.Status